### DISCUSSION TITLE
Change consumer options after initialization

### DISCUSSION THREAD

**Discussion by roi-orca** [bbc/sqs-consumer#369] [Feature Request]
Can we add the ability to change several options after initialization?
For example, I would like to change the visibility timeout based on runtime configuration.
Thanks!

**Comment by nicholasgriffintn** [comment]
This is something that we have considered but is potentially something that could be a bit messy, so something to think about.

We'd want to think about how we add functionality for changing the options and validating the options.

We'd also want to question what options should be available to change.

**Reply by roi-orca** [reply]
Of course, some options are immutable.
Maybe we can log/emit an event whether a new option is not valid and keep the old one active.
If we can implement some of the mutable options for the first phase such as the visibility timeout it will be great!
Thanks again!

**Reply by nicholasgriffintn** [reply]
Yeah, visibility timeout definitely makes a lot of sense, so I've raised a feature request for you here: https://github.com/bbc/sqs-consumer/issues/370.

I think that's possibly the best place to start, we may want to do similar after that has been completed for things like `batchSize` and `waitTimeSeconds`, which effect the options sent to SQS but not the options used by the Consumer itself.
